### PR TITLE
Feature/stip cell tooltip max width

### PR DESCRIPTION
**Pull request by lab900-gino-vanhoudt**
New minor version 3.1.0

patch notes: 
- Lab900TableComponent now has an input `[maxColumnWidth]="'200px'"`
  this ensures all columns have a max-width, with text-overflow, and a tooltip displaying the full content in the case of text-overflow
- TableCells now have an updated cellTooltip input, and a new cellMaxWidth input:
```typescript
 public tableCells: TableCell[] = [
    {
      key: 'nameLong',
      label: 'Long name',
      cellTooltip: { 
        text: (data) => data.nameLong, 
        onlyOnOverflow: true, 
        matTooltipOptions: { matTooltipPosition: 'left' }
      },
      cellMaxWidth: '300px', // overrides the maxColumnWidth on the table
    },
```
this tooltip will always show, unless you enable `onlyOnOverflow` 
this tooltip also overrides the default one from the `Lab900TableComponent` (in case a `maxColumnWidth` was set)
`cellMaxWidth` can be defined here too, overwriting `maxColumnWidth` from the `Lab900TableComponent`

## screenshots
hover over "overflown" line
image
hover over "non-overflown" line
image

### CODE REVIEW

File: lib/src/lib/table/models/table-cell-tooltip.model.ts [hunk 1]
```diff
@@ -0,0 +1 @@
...
   * Currently ony supports positioning option of matTooltip
   */
  matTooltipOptions?: {
    matTooltipPosition: TooltipPosition;
  };
}

export interface MatTooltipOptions {
  tooltipPosition?: TooltipPosition;
}
```
Comment (lab900-senn-lambrecht) [inline lib/src/lib/table/models/table-cell-tooltip.model.ts:1]: Something strange is happening here I think?

Comment (lab900-gino-vanhoudt) [reply lib/src/lib/table/models/table-cell-tooltip.model.ts]: @lab900-senn-lambrecht fixed, this also fixed the other linting stuff

File: lib/src/lib/table/models/table-cell-tooltip.model.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  /**
+   * If true, tooltip will only show when it the content of the cell is cut off (by setting TableCell.cellMaxWidth
+   * value)
+   * @Default false
+   */
+  onlyOnOverflow?: propValue<T, boolean>;
+  /**
+   * Currently ony supports positioning option of matTooltip
+   */
+  matTooltipOptions?: {
```
Comment (lab900-frederico-teixeira) [inline lib/src/lib/table/models/table-cell-tooltip.model.ts:18]: do we want to append `mat` to the name, what if we change completely the tooltip lib?

**Review by lab900-senn-lambrecht — APPROVED** [review]

File: lib/src/lib/table/models/table-cell-tooltip.model.ts (hunk 2 above)
Comment (lab900-gino-vanhoudt) [reply lib/src/lib/table/models/table-cell-tooltip.model.ts]: fixed

**Pull request merged**